tests/unit/pillar_test.py

```diff
@@ -130,38 +130,6 @@ def test_topfile_order(self, Matcher, get_file_client):
         pillar = salt.pillar.Pillar(opts, grains, 'mocked-minion', 'base')
         self.assertEqual(pillar.compile_pillar()['ssh'], 'foo')
 
-    @patch('salt.pillar.salt.fileclient.get_file_client', autospec=True)
```

Comment:
Commit 4af5b5c initially added the merge_lists=bool argument which caused the regression noted in #29601. This simply would set it back to True, as if it wasn't present to begin with and instead adds a new master_config opt of pillar_merge_lists (default True). In this way, it fixes the initially regression while still allowing someone to opt to not merge the lists by setting pillar_merge_lists: False in /etc/salt/master